app/lib/shared/task_scheduler.dart

```diff
@@ -74,3 +70,111 @@ class Task {
   @override
   int get hashCode => version.hashCode;
 }
+
+/// A pull-based interface for accessing events from multiple streams, in the
+/// priority order of the streams provided.
+///
+/// Inspired by Iterator and package:async's StreamQueue.
+class PrioritizedAsyncIterator<T> {
```

Comment:
Would it be worth having some sanity/unit tests for this class? Seems non-trivial